### PR TITLE
fixed circular includes in matrix classes

### PR DESCRIPTION
**Pull request by fmoessbauer**
removed circular includes of `dash/Container.h` and updated missing includes

### CODE REVIEW

File: dash/include/dash/matrix/MatrixRef.h [hunk 1]
```diff
@@ -9,7 +9,9 @@
 #include <dash/Pattern.h>
 #include <dash/GlobRef.h>
 #include <dash/HView.h>
-#include <dash/Container.h>
+
+#include <dash/view/Local.h>
```
Comment (fuchsto) [inline dash/include/dash/matrix/MatrixRef.h]: Are these required? The implementation of `dash::Matrix` does not depend on view expressions, does it?

Comment (fmoessbauer) [reply dash/include/dash/matrix/MatrixRef.h]: Unfortuately there are many many cross dependencies and the easiest way was to separate the includes to specific files. And yes, this is required.

Comment (fuchsto) [reply dash/include/dash/matrix/MatrixRef.h]: What for? Which definition would be missing otherwise?

Comment (fmoessbauer) [reply dash/include/dash/matrix/MatrixRef.h]: Just `dash/Sub.h` is sufficient, as `local` and `sub` are used.

**Review by devreal — APPROVED** [review]
Looks fine to me

**Pull request merged**